### PR TITLE
feat: user reading footprint

### PR DESCRIPTION
**Pull request by Aylie-Chou**
# Issue
[[會員制] 我的閱讀 - 造訪紀錄、收藏後端實作
](https://app.asana.com/0/1203699264568808/1206262043331068/f)

# Note
- add `/v2/user/:id/analytics/reading-footprint` enpoint (`POST`, `GET`)
- add `analytics` controller & storage (separate from membership)
- add doc for new added endpoints
- add test cases for new added endpoints
- add migrations for new added table `users_posts_reading_footprints`
  - schema at [notion doc](https://www.notion.so/twreporter/users_posts_reading_footprints-fb7b6c75b7e841b9a44129f35b20031c?pvs=4)

# Dependency
N/A

### CODE REVIEW

File: storage/analytics.go [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+		tx.Rollback()
+		return false, errors.Wrap(err, fmt.Sprintf("failed to check user reading footprint existence (user_id: %s, post_id: %s)", userID, postID))
+	}
+	// Directly return if record exist
+	if count != 0 {
+		tx.Rollback()
+		return true, nil
+	}
+
+	// Add the user read post record
```
Comment (liruchen32) [inline storage/analytics.go:131]: user reading foot print

File: storage/analytics.go [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+	tx := gs.db.Begin() // Start the transaction
+
+	// Check if the user exists
+	var count int64
+	if err := tx.Model(&models.UsersPostsReadingFootprint{}).Where("user_id = ? AND post_id = ?", userID, postID).Count(&count).Error; err != nil {
+		tx.Rollback()
+		return false, errors.Wrap(err, fmt.Sprintf("failed to check user reading footprint existence (user_id: %s, post_id: %s)", userID, postID))
+	}
+	// Directly return if record exist
+	if count != 0 {
```
Comment (liruchen32) [inline storage/analytics.go:126]: do we need to update `updated_at` if record exist

**Review by liruchen32 — COMMENTED** [review]
Just a small question and comment issue

File: storage/analytics.go (hunk 1 above)
Comment (Aylie-Chou) [reply storage/analytics.go]: fixed ([bc619b6](https://github.com/twreporter/go-api/pull/658/commits/bc619b6323638af4188e0f9dd58d6301c38f6cae))

File: storage/analytics.go (hunk 2 above)
Comment (Aylie-Chou) [reply storage/analytics.go:128]: yes !
fixed in commit [bc619b6](https://github.com/twreporter/go-api/pull/658/commits/bc619b6323638af4188e0f9dd58d6301c38f6cae)

**Review by liruchen32 — APPROVED** [review]
LGTM

**Pull request merged**